jina/drivers/craft.py

```diff
@@ -45,6 +45,8 @@ class SegmentDriver(CraftDriver):
 
     def __init__(self, first_chunk_id: int = 0, random_chunk_id: bool = True,
                  level_names: List[str] = None, *args, **kwargs):
+        # Important to do the overwrite before calling `super()__init__(...)`
+        kwargs.setdefault('traversal_paths', ['r', 'c'])
```

Comment:
isn't it more natural to simply pass this as named argument to the super().__init__()?

Reply:
nope, because you want to be able to even overwrite it from the .yml files. This is just setting the defaults for the specific driver.

Reply:
`setdefault` will only set it, if it is not yet set.

Reply:
Ah wait, you mean I could do:
```
def __init__(self, ..., traversal_paths=['r', 'c'], ...):
    super().__init__(traversal_paths=traversal_paths)
```
Yes that is better, I agree!

Reply:
ah good point! I just learned something!!!

Reply:
I did not mean that exactly, but I guess we had a good outcome, so yes it seems good!